livesync/indico_livesync/simplify.py

```diff
@@ -44,23 +53,80 @@ def process_records(records):
             continue
         if record.change == ChangeType.created:
             assert record.type != EntryType.category
-            changes[record.object_ref] |= SimpleChange.created
+            changes[record.object] |= SimpleChange.created
         elif record.change == ChangeType.deleted:
             assert record.type != EntryType.category
-            changes[record.object_ref] |= SimpleChange.deleted
+            changes[record.object] |= SimpleChange.deleted
         elif record.change in {ChangeType.moved, ChangeType.protection_changed}:
-            for ref in _cascade(record):
-                changes[ref] |= SimpleChange.updated
+            cascaded_records.add(record)
         elif record.change == ChangeType.data_changed:
             assert record.type != EntryType.category
-            changes[record.object_ref] |= SimpleChange.updated
+            changes[record.object] |= SimpleChange.updated
+
+    for obj in _process_cascaded(cascaded_records):
+        changes[obj] |= SimpleChange.updated
 
     return changes
 
 
-def _cascade(record):
-    yield record.object_ref
-    for subrecord in record.iter_subentries():
-        if subrecord.object:
-            for item in _cascade(subrecord):
-                yield item
+def _process_cascaded(records):
+    category_prot_records = {rec.category_id for rec in records if rec.type == EntryType.category
+                             and rec.change == ChangeType.protection_changed}
+    category_move_records = {rec.category_id for rec in records if rec.type == EntryType.category
+                             and rec.change == ChangeType.moved}
+
+    event_records = {rec.event_id for rec in records if rec.type == EntryType.event}
+    session_records = {rec.session_id for rec in records if rec.type == EntryType.session}
+    contribution_records = {rec.contribution_id for rec in records if rec.type == EntryType.contribution}
+    subcontribution_records = {rec.subcontribution_id for rec in records if rec.type == EntryType.subcontribution}
+
+    changed_events = set()
+    changed_contributions = set()
+    changed_subcontributions = set()
+
+    category_prot_records -= category_move_records  # A move already implies sending the whole record
+
+    # Protection changes are handled differently, as there may not be the need to re-generate the record
+    if category_prot_records:
+        for categ in Category.find(Category.id.in_(category_prot_records)):
+            cte = categ.get_protection_parent_cte()
+            # Update only children that inherit
+            inheriting_categ_children = (Event.query
+                                         .join(Category)
+                                         .join(cte, (Category.id == cte.c.id) & (cte.c.protection_parent == categ.id)))
```

Comment:
Do you actually need to `.join(Category)` here? Why not use `Event.category_id` in this filter instead